chrome/content/zotero/xpcom/zotero.js

```diff
@@ -381,7 +381,26 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 					return false;
 				}
 			}
-			
+			if(Zotero.File.isDropboxDirectory(dataDir.path) && Zotero.Prefs.get('warnDropboxDataDir')){
+				let check = {value: false};
+				ps = Services.prompt;
+				index = ps.confirmEx(null,
+					Zotero.getString('dataDir.existingDirInDropbox.title'),
+					Zotero.getString('dataDir.existingDirInDropbox.text', Zotero.appName),
+					ps.STD_YES_NO_BUTTONS,
+					null, null, null,
+					Zotero.getString('general.dontShowWarningAgain'),
+					check);
+
+					// Yes - display dialog.
+					if (index == 0) {
+						Zotero.chooseZoteroDirectory(true);
+					}
+					if(check.value === true) {
+						Zotero.Prefs.set('warnDropboxDataDir', false);
+					}
```

Comment:
Use this style to avoid mistakes like that:

```
var index = ps.confirmEx(
	null,
	dialogTitle,
	dialogText,
	buttonFlags,
	null,
	null,
	moreInfoCallback ? Zotero.getString('general.help') : null,
	null,
	{}
);
```